students/Chao/session04/trigram.py

```diff
@@ -1,5 +1,71 @@
 #!/usr/bin/env python
 
+import string
+import random
+
 # Create an empty dictionary
 trig = {}
+# An empty list for word pool
+wpool = []
+
+def build_pool(inf):
+    """ Build a pool of words from the input text file """
+    with open(inf, 'r') as lines:
+        # strip the line, make it all lower cases, remove punctuations, then make a new list
+        for line in lines:
+            wpool.extend(line.strip().lower().translate(str.maketrans('', '', string.punctuation)).split())
+
+        # change individual 'i' with 'I'
+        for wordi in range(len(wpool)):
+            if wpool[wordi] == 'i':
+                wpool[wordi] = 'I'
+
```

Comment:
Note that I is not the only word that may be capitalized in the middle of a sentence. It may make more sense to be more careful about what words you turn to lower case, rather than try to add back in capitalization for proper nouns. Note also that you are going back over the entire list to do this, which could get costly with a large text file.